File: lib/startup/cleanupFtpFolder.js
/*
 * Copyright (c) 2014-2021 Bjoern Kimminich.
 * SPDX-License-Identifier: MIT
 */

const glob = require('glob')
const path = require('path')
const fs = require('fs-extra')
const logger = require('../logger')

const cleanupFtpFolder = () => {
  glob(path.join(__dirname, '../../ftp/*.pdf'), (err, files) => {
    if (err) {
      logger.warn('Error listing PDF files in /ftp folder: ' + err.message)
    } else {
      files.forEach(filename => {
        fs.remove(filename)
      })
    }
  })
}

module.exports = cleanupFtpFolder

File: lib/startup/cleanup_ftp_folder.py
import os
import glob
import logging

logger = logging.getLogger(__name__)

def cleanup_ftp_folder():
    try:
        files = glob.glob('ftp/*.pdf')
        for filename in files:
            os.remove(filename)
    except Exception as e:
        logger.warning(f"Error listing PDF files in /ftp folder: {str(e)}")
